packages/web/src/components/edit-track/EditTrackForm.tsx

```diff
@@ -186,6 +210,52 @@ const TrackEditForm = (
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [getArtworkUrl(updatedArtwork), setArtworkValue])
 
+  const onClickReplace = useCallback(
+    async (file: File) => {
+      const processedFiles = await Promise.all(
+        processFiles([file], (name, reason) => {
+          console.error('Failed to process file', name, 'due to', reason)
+          // todo: show error state
+        })
+      )
+
+      if (processedFiles?.length > 0) {
+        const newFile = processedFiles[0]!
+
+        if (isUpload && !isTitleDirty) {
+          setTitle(newFile.metadata.title.split('.').shift())
+        }
+        if (isUpload && !isArtworkDirty && newFile.metadata.artwork.file) {
+          setArtworkValue(newFile.metadata.artwork)
+        }
+        setTrackValue(newFile)
+        setOrigFilename(newFile.metadata.orig_filename)
+      }
+    },
+    [
+      isArtworkDirty,
+      isTitleDirty,
+      isUpload,
+      setArtworkValue,
+      setOrigFilename,
+      setTitle,
+      setTrackValue
+    ]
+  )
+
+  const onClickDownload = useCallback(() => {
+    if (!track.file) return
+
+    const url = URL.createObjectURL(track.file)
+    const link = document.createElement('a')
+    link.href = url
+    link.download = track.file.name || 'download'
+    document.body.appendChild(link)
+    link.click()
+    document.body.removeChild(link)
+    URL.revokeObjectURL(url)
```

Comment:
I know I added this, but do you know of any cleaner way to do this?

I also noticed that this didn't pop up a choose save location window, it just chucks it into downloads. Maybe can check w/Sammie which is more desirable

Reply:
yea i hadn't looked, but I can check in with Sammie and see how she wants it to work